zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java

```diff
@@ -85,13 +86,14 @@ private void initialiseIndices() {
     }
 
     final var existingIndexNames =
-        searchEngineClient.getMappings(allIndexNames(), MappingSource.INDEX).keySet();
+        Collections.synchronizedSet(
+            searchEngineClient.getMappings(allIndexNames(), MappingSource.INDEX).keySet());
 
     LOG.info(
         "Found '{}' existing indices. Create missing index templates based on '{}' descriptors.",
         existingIndexNames.size(),
         indexTemplateDescriptors.size());
-    indexDescriptors.stream()
+    indexDescriptors.parallelStream()
```

Comment:
:wrench: `.parallelStream()` will spawn tasks in the static `ForkJoinPool.commonPool`. In general ForkJoinPool should not be used for blocking tasks, but:
- the max parallelism by default is 256 (which is enough for these tasks and to not block others using it)
- it's done only at startup (if needed)

Do you know if we use already this pool? If we are not then it will be allocated here and then probably stay alive. 

If  we are not using one already, we can create a local executor that we close the schema is created.

Reply:
Thanks @entangled90 for the input! I knew there was something not optimal with the fork-join pool, but didn't remember what it was. But anyhow it was mostly more a POC whether it helps :) 


We could also think about virtual threads maybe? 🤔 We have already an executor created here for the incident notification https://github.com/camunda/camunda/blob/main/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/DefaultExporterResourceProvider.java#L152

Reply:
I was under the impression that until this [JEP](https://openjdk.org/jeps/491) lands it's a bit risky to use virtual threads, but i'm not up to date on those decisions.

Anyway for this task it should be ok, even if they lock the native thread it's still a better parallelism than 1 :-D

Reply:
Ok let me try this